pkg/helper/task_runner.go

```diff
@@ -0,0 +1,52 @@
+package helper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+type task struct {
+	Name string
+	Run  func(interface{}) error
```

Comment:
Why is the interface{} parameter added as a parameter of the Run method?

Reply:
A context.

Some context that tasks can share to pass messages or objects.

It was used initially, but after a refactor, all tasks belong to the same type and they can share information using the type receiver. 

I left there for being more flexible for future use. But can be removed now and added later if needed.

Reply:
No need for now